src/main/scala/ch/epfl/bluebrain/nexus/admin/organizations/Organizations.scala

```diff
@@ -1,16 +1,183 @@
+
+  /**
+    * Deprecate an organization.
+    *
+    * @param id     ID of the organization to update
+    * @param rev    latest known revision
+    * @param caller identity of the caller performing the operation
+    * @return       metadata about the organization
+    */
+  def deprecate(id: UUID, rev: Long)(implicit caller: Identity): F[OrganizationMetaOrRejection] =
+    evaluate(DeprecateOrganization(id, rev, clock.instant(), caller))
+
+  /**
+    * Fetch an organization.
+    *
+    * @param id ID of the organization to fetch
+    * @return   organization and metadata if it exists, None otherwise
+    */
+  def fetch(id: UUID): F[Option[ResourceF[Organization]]] =
+    agg
+      .currentState(id.toString)
+      .map(stateToResource)
+
+  /**
+    * Fetch an organization by revision
+    * @param id   ID of the organization to fetch
+    * @param rev  revision to fetch
+    * @return     organization and metadata if it exists, None otherwise
+    */
+  def fetch(id: UUID, rev: Long): F[Option[ResourceF[Organization]]] =
+    agg
+      .foldLeft[OrganizationState](id.toString, Initial) {
+        case (state, event) if event.rev <= rev => next(state, event)
+        case (state, _)                         => state
+      }
+      .map(stateToResource)
+
+  private def evaluate(cmd: OrganizationCommand): F[OrganizationMetaOrRejection] =
+    agg
+      .evaluateS(cmd.id.toString, cmd)
+      .map(
+        _.flatMap {
+          case c: Current => Right(c.toResourceMetadata)
+          case Initial    => Left(OrganizationUnexpectedState(cmd.id))
+        }
+      )
+
+  private def stateToResource(state: OrganizationState): Option[ResourceF[Organization]] = state match {
+    case Initial    => None
+    case c: Current => Some(c.toResource)
+  }
+
+}
+object Organizations {
+
+  /**
+    * Construct ''Organization'' wrapped on an ''F'' type based on akka clustered [[Aggregate]].
+    */
+  def apply[F[_]: ConcurrentEffect](implicit cl: Clock = Clock.systemUTC(),
+                                    ac: AppConfig,
+                                    sc: AkkaSourcingConfig,
+                                    as: ActorSystem,
+                                    mt: ActorMaterializer): F[Organizations[F]] = {
+    val aggF
+      : F[Aggregate[F, String, OrganizationEvent, OrganizationState, OrganizationCommand, OrganizationRejection]] =
+      AkkaAggregate.sharded(
+        "organizations",
+        Initial,
+        next,
+        evaluate[F],
+        PassivationStrategy.immediately[OrganizationState, OrganizationCommand],
+        RetryStrategy.never,
+        sc,
+        ac.cluster.shards
+      )
+
+    aggF.map(new Organizations(_))
+  }
+
+  /**
+    * Construct ''Organizations'' wrapped on an ''F'' type based on an in memory [[Aggregate]].
+    */
+  def inMemory[F[_]: ConcurrentEffect](implicit cl: Clock, http: HttpConfig): F[Organizations[F]] = {
```

Comment:
I moved this constructor in the test itself, I don't think we'll ever need it here.